server/mlabns/util/fqdn_rewrite.py

```diff
@@ -0,0 +1,102 @@
+"""Rewrites raw M-Lab FQDNs to apply post-processing or annotations."""
+
+import logging
+
+from mlabns.util import message
+
+
+def rewrite(fqdn, address_family, tool_id):
+    """Rewrites an FQDN to add necessary annotations and special-casing.
+
+    Performs the following rewrites on an FQDN:
+    * Adds a v4/v6 annotation if the client requested an explicit address family
+    * Applies a workaround to fix FQDNs for NDT-SSL queries.
+
+    Args:
+        fqdn: A tool FQDN with no address family specific annotation.
+        address_family: The address family for which to create the FQDN or None
+            to create an address family agnostic FQDN.
+        tool_id: Name of tool associated with the FQDN (e.g. 'ndt_ssl').
+
+    Returns:
+        FQDN after rewriting to apply all modifications to the raw FQDN.
+    """
+    rewritten_fqdn = _apply_af_awareness(fqdn, address_family)
```

Comment:
Ah, okay, I see what you're saying.

How much work would it be for us to publish AF-decorated RRs for NDT-SSL?

I can apply another special case in `fqdn_rewrite.rewrite()` but I'd like to limit special casing if possible because it makes it much easier for bugs to crop up when we forget that some other piece of the system assumes no special casing. So if this is something we're going to end up doing in the future, I'd prefer to do it now.

I actually know nothing about why we publish AF-decorated RRs in the first place. I just found that behavior defined in the mlab-ns spec and tried to make sure we preserved it.